showcase/src/app/demo/pretty-print/pretty-print.component.ts

```diff
@@ -0,0 +1,152 @@
+			path: "pretty-print/scss",
+			rawData: [
+				"$font-stack: Helvetica, sans-serif; $primary-color: #333; body { font: 100% $font-stack; color: $primary-color; }"
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddHtml(): void {
+		const fileType: FileType = {
+			format: "html",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.HTML",
+			path: "pretty-print/html",
+			rawData: [
+				"<!DOCTYPE html><html><head><style>body {background-color: powderblue;}h1{color: blue;}p{color: red;}",
+				"</style></head><body><h1>This is a heading</h1><p>This is a paragraph.</p></body></html>"
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddJavascript(): void {
+		const fileType: FileType = {
+			format: "javascript",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.JAVASCRIPT",
+			path: "pretty-print/javascript",
+			rawData: [
+				"function calculateData(seed, operationFn) {",
+				"var data = operationFn(seed);",
+				"if (!data){",
+				"data = 'could not calculate data';",
+				"}",
+				"return data;",
+				"}"
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddTypescript(): void {
+		const fileType: FileType = {
+			format: "typescript",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.TYPESCRIPT",
+			path: "pretty-print/typescript",
+			rawData: [
+				"function calculateData(seed:any, operationFn:Function):any {",
+				"var data:any = operationFn(seed);",
+				"if (!data){",
+				"data = 'could not calculate data';",
+				"}",
+				"return data;",
+				"}"
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddJson(): void {
+		const fileType: FileType = {
+			format: "json",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.JSON",
+			path: "pretty-print/json",
+			rawData: [
+				'{"menu": { "id": "file", "value": "File",',
+				'"menuitem": [{"value": "New", "onclick": "CreateNewDoc()"},',
+				'{"value": "Open", "onclick": "OpenDoc()"},',
+				'{"value": "Close", "onclick": "CloseDoc()"}]}}'
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddXml(): void {
+		const fileType: FileType = {
+			format: "xml",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.XML",
+			path: "pretty-print/xml",
+			rawData: [
+				'<menu id="file" value="File"><menuitem value="New" onclick="CreateNewDoc()" />',
+				'<menuitem value="Open" onclick="OpenDoc()" />',
+				'<menuitem value="Close" onclick="CloseDoc()" /></menu>'
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+
+	private AddSql(): void {
+		const fileType: FileType = {
+			format: "sql",
+			title: "SHOWCASE.DEMO.PRETTY-PRINT.SQL",
+			path: "pretty-print/sql",
+			rawData: [
+				"SELECT DISTINCT Name FROM Production.Product AS p WHERE EXISTS (SELECT * ",
+				"FROM Production.ProductModel AS pm WHERE p.ProductModelID = pm.ProductModelID ",
+				"AND pm.Name LIKE 'Long-Sleeve Logo Jersey%')"
+			].join("")
+		};
+		this.fileTypes.push(fileType);
+	}
+}
+
+export interface FileType {
+	format: string;
```

Comment:
You can refine this type to the different options you will expect: "javascript" | "sql" | ...